Shopify/cli
# add structured logging

### WHY are these changes introduced?

Fixes [#371](https://github.com/Shopify/shopify-cli-extensions/issues/371)
Enhance the extension server logging replacing the human readable messages by structured Json logs.
The problem is describe further in this [document](https://docs.google.com/document/d/1EaxKuiDOHuukkqayNr_Ot-7JcAwomjhDYnLcwfNuLko/edit#)

### WHAT is this pull request doing?
- Created a log structure with semantically descriptive context
- disable standard logging to use the new structured logs

### How to test your changes?
- create a new go binary
- run any of the supported flows using the binary (scaffold or dev)
- check the output logs
